server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -218,81 +214,56 @@ private synchronized void syncSegments(boolean isRetry) {
-                            throw e;
-                        }
-                        localSegmentsPostRefresh.addAll(segmentCommitInfos.files(true));
-                        segmentInfosFiles.stream()
-                            .filter(file -> !file.equals(latestSegmentInfos.get()))
-                            .forEach(localSegmentsPostRefresh::remove);
-
-                        // Create a map of file name to size and update the refresh segment tracker
-                        updateLocalSizeMapAndTracker(localSegmentsPostRefresh);
-                        CountDownLatch latch = new CountDownLatch(1);
-                        ActionListener<Void> segmentUploadsCompletedListener = new LatchedActionListener<>(new ActionListener<>() {
-                            @Override
-                            public void onResponse(Void unused) {
-                                boolean shouldRetry = true;
-                                try {
-                                    // Start metadata file upload
-                                    uploadMetadata(localSegmentsPostRefresh, segmentInfos);
-                                    clearStaleFilesFromLocalSegmentChecksumMap(localSegmentsPostRefresh);
-                                    onSuccessfulSegmentsSync(
-                                        refreshTimeMs,
-                                        refreshClockTimeMs,
-                                        refreshSeqNo,
-                                        lastRefreshedCheckpoint,
-                                        checkpoint
-                                    );
-                                    // At this point since we have uploaded new segments, segment infos and segment metadata file,
-                                    // along with marking minSeqNoToKeep, upload has succeeded completely.
-                                    shouldRetry = false;
-                                } catch (Exception e) {
-                                    // We don't want to fail refresh if upload of new segments fails. The missed segments will be re-tried
-                                    // in the next refresh. This should not affect durability of the indexed data after remote trans-log
-                                    // integration.
-                                    logger.warn("Exception in post new segment upload actions", e);
-                                } finally {
-                                    doComplete(shouldRetry);
-                                }
+                    // Create a map of file name to size and update the refresh segment tracker
+                    updateLocalSizeMapAndTracker(localSegmentsPostRefresh);
+
+                    // Create a map of file name to size and update the refresh segment tracker
+                    updateLocalSizeMapAndTracker(localSegmentsPostRefresh);
```

Comment:
Looks like an unintentional duplicate call here?

Reply:
Yes, would have got added as part of resolving merge conflicts. Thanks for catching this.